.github/workflows/api-deploy-production.yml

```diff
@@ -22,6 +22,22 @@ jobs:
               with:
                   fetch-depth: 0
 
+            - name: Add SAML package
```

Comment:
I have not explored much but, I think we can use Reusable workflows here? Put all of this in a common workflow and use that?

Reply:
Yep, good shout. 

They're kinda cool, but a bit annoying. 

The main issues I have: 

 * You seemingly have to provide a git reference for the workflow, you can't just use the workflow from the current commit (unless you manually set the branch). I tried using `${GITHUB_SHA}` and that just complained that it wasn't valid. I've just used `main` for now but we may want to switch it to a sha once this PR is merged. 
 * The way in which secrets work is a bit confusing. You have to provide the github environment through to the reusable job but then provide the secrets to the reusable workflow from the calling workflow. 

There's a chance we may be better off using local actions instead. I'll have a look into that quickly now but you can have a look at what's there in answer to your comment about reusable workflows at least.

Reply:
> You seemingly have to provide a git reference for the workflow, you can't just use the workflow from the current commit 

That seems a bit annoying. 

>you can have a look at what's there in answer to your comment about reusable workflows at least 

Yep, that's what I was talking about

Reply:
Ok, I've implemented it using "local composite actions" instead. The documentation is pretty terrible but I think it's a bit neater. 

It solves the 2 issues I mentioned above but it comes with a small amount less code reuse: 

 * We have to checkout the code in the 'calling workflow'
 * We have to define the `container` & `runs-on` variables in the calling workflow

I still think it's a bit neater but I'm interested in your thoughts @gagantrivedi and @dabeeeenster

Reply:
For reference, here are the differences between the two options: https://github.com/Flagsmith/flagsmith/pull/667/commits/dfda7d640acbe538d4e63862bf7089f3103407ad